framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceJavaParserStorage.java

```diff
@@ -706,64 +706,50 @@ public void processClass(ClassTree javacTree, AnnotationDeclaration javaParserNo
           public void processNewClass(NewClassTree javacTree, ObjectCreationExpr javaParserNode) {
             ClassTree body = javacTree.getClassBody();
             if (body != null) {
-              // elementFromTree returns null instead of crashing when no element exists
-              // for the class tree, which can happen for certain kinds of anonymous
-              // classes, such as Ordering$1 in PolyCollectorTypeVar.java in the
-              // all-systems test suite.  addClass(ClassTree) in the then branch just
-              // below assumes that such an element exists.
-              Element classElt = TreeUtils.elementFromDeclaration(body);
-              if (classElt != null) {
-                addClass(body, null);
-              } else {
-                // If such an element does not exist, compute the name of the class,
-                // instead.  This method of computing the name is not 100% guaranteed to
-                // be reliable, but it should be sufficient for WPI's purposes here: if
-                // the wrong name is computed, the worst outcome is a false positive
-                // because WPI inferred an untrue annotation.
-                @BinaryName String className;
-                if ("".contentEquals(body.getSimpleName())) {
-                  @SuppressWarnings("signature:assignment") // computed from string concatenation
-                  @BinaryName String computedName =
-                      javaParserClass.getFullyQualifiedName().get() + "$" + ++innerClassCount;
-                  className = computedName;
-                } else {
-                  @SuppressWarnings("signature:assignment") // computed from string concatenation
-                  @BinaryName String computedName =
-                      javaParserClass.getFullyQualifiedName().get()
-                          + "$"
-                          + body.getSimpleName().toString();
-                  className = computedName;
-                }
-                addClass(body, className, null);
-              }
+              addClass(body, null);
             }
           }
 
           /**
            * Creates a wrapper around the class for {@code tree} and stores it in {@code
            * sourceAnnos}.
            *
-           * <p>This method assumes that there is an Element corresponding to {@code tree} if {@code
-           * classNameKey} is null.
+           * <p>This method computes the name of the class and uses it as the key for {@code
+           * classToAnnos}
```

Comment:
This sentence is only true when the element corresponding to `tree` is null, right? I think that should be in this javadoc.